---
title: "snippy analysis"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
```

```{r}
fa_filest <- read.table("~/Documents/2021-Fiji-Ct_paper/Published_CT_Genomes/fa_filest.list", quote="\"", comment.char="")
Tip_Labels_BigTree <- read.delim("~/Documents/2021-Fiji-Ct_paper/Published_CT_Genomes/Tip_Labels_BigTree.txt") %>%
  filter(grepl("urogenital",Lineage)) %>%
  filter(grepl("^\\D",taxa)) %>%
  mutate(Clade = ifelse(grepl("Non",Lineage),"T2","T1"))
CT_Mapping_05_27 <- read.delim("~/Documents/2021-Fiji-Ct_paper/Published_CT_Genomes/CT_Mapping_05_27.txt", header=FALSE) %>%
  select(filenm = V1,taxa = V2) %>%
  mutate(filenm = gsub(".fa|.fasta","",filenm)) %>%
  mutate(filenm = gsub("-WGS","",filenm))
T1T2 <- inner_join(Tip_Labels_BigTree,CT_Mapping_05_27, by = "taxa") %>%
  mutate(fa_filenm = paste0(filenm,".fa")) %>%
  filter(fa_filenm %in% fa_filest$V1)

```
Make snippy table
```{r}
filepa <- "/Users/timothyread/Documents/2021-Fiji-Ct_paper/Published_CT_Genomes/"
T1T2_tab <- T1T2 %>%
  mutate(taxa = paste0(Clade,"_",taxa)) %>% 
  select(taxa,fa_filenm) %>%
  filter(taxa != "T1_E-160") %>%
  filter(taxa != "T1_E-8873") %>%
  mutate(fa_filenm = paste0(filepa,fa_filenm))
write_tsv(T1T2_tab,"/Users/timothyread/Documents/2021-Fiji-Ct_paper/2022-02-T1T2-snippy/input.tab")
```
read output of snippy_core
```{r}
core_tab <- read.delim("~/Documents/2021-Fiji-Ct_paper/2022-02-T1T2-snippy/core.tab")
```

use if_all and across for filtering see https://dplyr.tidyverse.org/articles/colwise.html
```{r}
core_tab %>%
  filter(if_all(contains("T1"), ~ .x == REF)) %>%
  filter(if_all(contains("T2"), ~ .x != REF))
```
There are no SNPs that are conserved across all strains

```{r}
core_concord <- core_tab %>%
  mutate(across(contains("T1"), ~ .x == REF)) %>%
  mutate(across(contains("T2"), ~ .x != REF)) %>%
  rowwise() %>% 
  mutate(concordant_strains = sum(c_across(starts_with("T")))) %>% 
  ungroup()
```


```{r}
ggplot(core_concord,aes(concordant_strains)) +
  geom_histogram(binwidth = 1) +
  theme_bw()
```

```{r}
T1T2_snps_30 <- core_concord %>%
  filter(concordant_strains > 30) %>% .$POS
T1T2_snps_39 <- core_concord %>%
  filter(concordant_strains > 39) %>% .$POS
write_lines(T1T2_snps_30,"T1T2_snps_30.list")
write_lines(T1T2_snps_39,"T1T2_snps_39.list")
```

### Now repeat with denovo genomes

```{r}
denovo_genomes <- c("927C","26C","68C","1226V","192V","951C","908C","1145C","519V","107R","1077C","1079C","362C","98V","1201V","87C","227C","1106C","753C","968V","72V")
denovo_files <- read.table("~/Documents/2021-Fiji-Ct_paper/CT_denovo_assemblies/denovo_fa.list", quote="\"", comment.char="")
Tip_Labels_BigTree2 <- read.delim("~/Documents/2021-Fiji-Ct_paper/Published_CT_Genomes/Tip_Labels_BigTree.txt") %>%
  filter(grepl("urogenital",Lineage)) %>%
  filter(grepl("^\\d",taxa)) %>%
  mutate(Clade = ifelse(grepl("Non",Lineage),"T2","T1"))
T1T2v2 <- inner_join(Tip_Labels_BigTree2,CT_Mapping_05_27, by = "taxa") %>%
  mutate(fa_filenm = paste0(filenm,".fa")) %>%
  filter(taxa %in% denovo_genomes) %>%
  filter(fa_filenm %in% denovo_files$V1)

```

Make snippy table for v2
```{r}
filepa2 <- "/Users/timothyread/Documents/2021-Fiji-Ct_paper/CT_denovo_assemblies/"
T1T2v2_tab <- T1T2v2 %>%
  mutate(taxa = paste0(Clade,"_",taxa)) %>% 
  select(taxa,fa_filenm) %>%
  filter(taxa != "T1_E-160") %>%
  filter(taxa != "T1_E-8873") %>%
  mutate(fa_filenm = paste0(filepa2,fa_filenm))
write_tsv(T1T2v2_tab,"/Users/timothyread/Documents/2021-Fiji-Ct_paper/2022-02-T1T2-snippy/input.tab")
```
 Now process the core snp tables again
 
```{r}
core <- read.delim("~/Documents/2021-Fiji-Ct_paper/2022-02-T1T2-snippy/core.tab")
```
 
```{r}
core_concord2 <- core %>%
  mutate(across(contains("T1"), ~ .x == T1_1226V)) %>%
  mutate(across(contains("T2"), ~ .x != T1_1226V)) %>%
  rowwise() %>% 
  mutate(concordant_strains = sum(c_across(starts_with("T")))) %>% 
  ungroup()
```

```{r}
ggplot(filter(core_concord2,concordant_strains > 18),aes(POS)) +
  geom_histogram(binwidth = 1000) +
  theme_bw()
```


Attempt to remove recombinant regions based on gubbins

```{r}
CT_Gubbins <- read.delim("~/Documents/2021-Fiji-Ct_paper/Fiji_onlyTree_12_06_2021/CT_Gubbins_All_fiji_Only_aln_12_06.recombination_predictions_no_head.txt", header=FALSE) %>%
  select(V4,V5)
result <- vector()
for (i in 1:nrow(CT_Gubbins)) {
      result <- c(result,seq(CT_Gubbins$V4[i],CT_Gubbins$V5[i]))
}
result <- unique(result)
```

filter reuslts
```{r}
T1T2_denovo_snps <- core_concord2 %>%
  filter(concordant_strains > 17) %>%
  filter(!POS %in% result) %>% 
  .$POS

```

```{r}
write_lines(T1T2_denovo_snps,"T1T2_snps_denovo.list")
```

without filtering for recombinant regions
```{r}
T1T2_denovo_snps_v2 <- core_concord2 %>%
  filter(concordant_strains > 17) %>%
  .$POS
```


```{r}
write_lines(T1T2_denovo_snps_v2,"T1T2_snps_denovo_v2.list")
```
