---
title: "R Notebook"
output: html_notebook
---



```{r}
library(tidyverse)
library(Biostrings)
```


```{r}
Sub_1182R_potential_minors <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182R_potential_minors.tsv")
Sub_1182V_potential_minors <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182V_potential_minors.tsv")
Sub_1182C_potential_minors <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182C_potential_minors.tsv")
Sub_1182V_SNPs <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182V_potential_SNPs.tsv")
Sub_1182R_SNPs <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182R_potential_SNPs.tsv")
Sub_1182C_SNPs <- read.delim("~/Documents/2021-Fiji-Ct_paper/MAP_results/1182C_potential_SNPs.tsv")

```

```{r}
Sub_1182R_SNPs %>% 
  filter(Position %in% Sub_1182V_potential_minors$Position)
```

```{r}
partition_1182V_SNPs <- Sub_1182V_SNPs %>% 
  add_row(Sub_1182V_potential_minors) %>%
  mutate(pos1182R = as.factor(ifelse(Position %in% Sub_1182R_potential_minors$Position,"1182R_iSNP","other_SNP"))) %>%
  mutate(SNP_percent = round(Ref_cov/(Ref_cov + Aa + Gg + Cc + Tt)*100,digits=1)) %>%
  mutate(Highest_SNPs = as.factor(ifelse(SNP_percent < 2,1,0))) %>%
  mutate(Medium_SNPs= as.factor(ifelse(SNP_percent > 2 & SNP_percent < 21,1,0)))

Medium_1182V_SNPss <- partition_1182V_SNPs %>% 
  filter(Medium_SNPs == 1) %>%
  .$Position

High_1182V_SNPss <- partition_1182V_SNPs %>% 
  filter(Highest_SNPs == 1) %>%
  .$Position

table(partition_1182V_SNPs$pos1182R)
```

```{r}
    ggplot(partition_1182V_SNPs,aes(x=SNP_percent,fill = pos1182R)) + 
    geom_histogram(binwidth=1, alpha=0.2, position="identity") +
    ggtitle("Frequency pf 1182V SNPs by match to 1182R") +
    theme_bw()
```

From this result , it appears there are two populations in 1182V - one set that matches a sub population in 1182R and a set of ~ 250 SNPs.  Are these SNPs found in all strains maybe due to differences with the reference?



```{r}
    ggplot(partition_1182V_SNPs,aes(x=Position,fill = pos1182R)) + 
    geom_histogram(binwidth=20000, alpha=0.2, position="identity") +
    ggtitle("Chromosome position of 1182V SNPs by match to 1182R") +
    theme_bw()
```


```{r}
partition_1182C_SNPs <- Sub_1182C_SNPs %>% 
  add_row(Sub_1182C_potential_minors) %>%
  mutate(pos1182R = as.factor(ifelse(Position %in% Sub_1182R_potential_minors$Position,"1182R_iSNP","other_SNP"))) %>%
  mutate(medium_pos1182V = as.factor(ifelse(Position %in% Medium_1182V_SNPss,"1182V_medium_SNP","not_1182V_medium_SNP"))) %>%
  mutate(high_pos1182V = as.factor(ifelse(Position %in% High_1182V_SNPss,"1182V_high_SNP","not_1182V_high_SNP"))) %>%
  mutate(SNP_percent = round(Ref_cov/(Ref_cov + Aa + Gg + Cc + Tt)*100,digits=1))

ggplot(partition_1182C_SNPs,aes(x=SNP_percent,fill = pos1182R)) + 
    geom_histogram(binwidth=1, alpha=0.2, position="identity") +
    ggtitle("Frequency of 1182C SNPs by match to 1182R") +
    theme_bw()
```

```{r}
ggplot(partition_1182C_SNPs,aes(x=SNP_percent,fill = medium_pos1182V)) + 
    geom_histogram(binwidth=1, alpha=0.2, position="identity") +
    ggtitle("Frequency of 1182C SNPs by match to 1182V SNPs > 2-20%") +
    theme_bw()
```

```{r}
ggplot(partition_1182C_SNPs,aes(x=SNP_percent,fill = high_pos1182V)) + 
    geom_histogram(binwidth=1, alpha=0.2, position="identity") +
    ggtitle("Frequency of 1182C SNPs by match to 1182V SNPs < 2%") +
    theme_bw()
```


```{r}
 ggplot(partition_1182C_SNPs,aes(x=Position,fill = pos1182V)) + 
    geom_histogram(binwidth=20000, alpha=0.2, position="identity") +
    ggtitle("Chromosome position of 1182C SNPs by match to 1182V") +
    theme_bw()
```

```{r}
table(partition_1182C_SNPs$pos1182R)
```

# recombination sites (see Gubbins_analysis.Rmd)

Patition into 2 sets of SNPs based on not matching the 

```{r}
bedfile_1182V.2 <- read.delim("~/Documents/2021-Fiji-Ct_paper/Manuscript_Figures_data/Global_PhylogeneticTree/Snippy_alignment/1182V-2.bed", header=FALSE)
recomb_irange <- IRanges(start=bedfile_1182V.2$V2,end=bedfile_1182V.2$V3)
other_SNPs_1182C <- partition_1182C_SNPs %>%
  filter(pos1182V == "other_SNP" & pos1182R == "other_SNP") %>%
  .$Position 

shared_SNPs_1182C <- partition_1182C_SNPs %>%
  filter(pos1182V == "1182V_SNP" & pos1182R == "other_SNP") %>%
  .$Position 

sub1182C_other_snps_irange <- IRanges(start= other_SNPs_1182C ,width=1)
sub1182C_shared_snps_irange <- IRanges(start= shared_SNPs_1182C ,width=1)
```


```{r}
findOverlaps(recomb_irange,sub1182C_other_snps_irange)
```

```{r}
findOverlaps(recomb_irange,sub1182C_shared_snps_irange)
```
